stacks/services/GeneratorLambda.ts

```diff
@@ -1,20 +1,30 @@
 import { PolicyStatement } from "aws-cdk-lib/aws-iam";
-import { Bucket, Stack, Function } from "sst/constructs";
+import { Bucket, Stack, Function, Table } from "sst/constructs";
 
-export const createGeneratorLambda = (stack: Stack, siriSXUnvalidatedBucket: Bucket, disruptionsJsonBucket: Bucket) => {
+export const createGeneratorLambda = (stack: Stack, siriSXUnvalidatedBucket: Bucket, table: Table) => {
     const siriGenerator = new Function(stack, "cdd-siri-sx-generator", {
         functionName: `cdd-siri-sx-generator-${stack.stage}`,
         environment: {
+            TABLE_NAME: table.tableName,
             SIRI_SX_UNVALIDATED_BUCKET_NAME: siriSXUnvalidatedBucket.bucketName,
         },
         permissions: [
             new PolicyStatement({
-                actions: ["s3:GetObject"],
-                resources: [`${disruptionsJsonBucket.bucketArn}/*`],
+                resources: [`${siriSXUnvalidatedBucket.bucketArn}/*`],
+                actions: ["s3:PutObject"],
             }),
             new PolicyStatement({
-                actions: ["s3:PutObject"],
-                resources: [`${siriSXUnvalidatedBucket.bucketArn}/*`],
+                resources: [table.tableArn],
+                actions: ["dynamodb:Scan"],
+            }),
+            new PolicyStatement({
+                resources: [`${table.tableArn}/stream/*`],
+                actions: [
+                    "dynamodb:GetRecords",
+                    "dynamodb:DescribeStream",
+                    "dynamodb:GetShardIterator",
```

Comment:
GetSharedIterator ?

Reply:
It's to enable the function to examine the kinesis shard of the dynamodb stream